src/main/java/net/starschema/clouddb/jdbc/BQStatement.java

```diff
@@ -183,8 +188,8 @@ private ResultSet executeQueryHelper(String querySql, boolean unlimitedBillingBy
               && (qr.getTotalRows().equals(BigInteger.ZERO)
                   || (qr.getRows() != null
                       && qr.getTotalRows().equals(BigInteger.valueOf(qr.getRows().size()))));
-      // Don't look up the job if we have nothing else we need to do
-      if (!(fetchedAll || this.connection.isClosed())) {
```

Comment:
When is it necessary to query BI engine statistics after having fetched all results? Should that be a test case?

Reply:
> Some minor nits, but also seems like incomplete test coverage. Is that coming in a follow-up PR?
Unfortunately I am very limited in terms of tests as we don't currently have a test DB availiable for JDBC with BI Engine enabled.. I tests thanks to HellTool

Reply:
Actually contrary to cacheHit we only get the BiEngine results in the end. I'll double check

Reply:
Read it like this: Before we needed the referenced job only when fetchedAll was false because we still had to do some stuff with the job. Now even if everything was indeed fetched, we need the referencedJob in order to get the BiEngine Statistics